### DISCUSSION TITLE
The problem when adding Intel RealSense Camera in Isaac Sim

### DISCUSSION THREAD

**Discussion by ge79puv** [StanfordVL/BEHAVIOR-1K#1014] [Q&A]
Hey there,

now I use Isaac SIm 4.1.0, and I am trying to add a RGB-D sensor to the wirst of my manipulation robot. I found there is already Realsense 455 intergrated in the software, so I tried to add this directly follwed the instruction in https://behavior.stanford.edu/omnigibson/tutorials/custom_robot_import.html. However, there is an error about the "missing reference found" after I added it in the software. And I also cannot find the corresponding usd file mentioned in https://docs.omniverse.nvidia.com/isaacsim/latest/features/environment_setup/assets/usd_assets_sensors.html. Could you tell me how to solve this problem?

Moreover, it's better for me to add Realsense D435i sensor into the simulator, because there is already one available in the labor. Then, I import the urdf file of  Realsense D435i in the Isaac Sim successfully, but there is only proprio value returned from get_obs() function. It seems that there are more steps that I do not know. Could you help me to complete that?

Finally, I found that I can directly add a Camera for the robot, and it can also return the rgb and depth value when I use get_obs() function. However, I am worried about the intrinsic parameters would be different from D435i camera. I notice there are some editable parameters in Isaac Sim for the camera, but can that cover all the relevant parameters of Realsense D435i?

Thanks a lot for your help in advance!

Best regards,
Yunfeng

**Comment by hang-yin** [comment]
Duplicate question from https://github.com/StanfordVL/OmniGibson/discussions/1007

**Reply by ge79puv** [reply]
Hi @hang-yin ,

thanks for your rapid and careful reply! I will not try to add RealSense D455 in Isaac Sim. But I think my question is a little bit different from https://github.com/StanfordVL/OmniGibson/discussions/1004. I am wondering if I add a Camera (maybe equal to a RGB-D camera? Please correct me if I am wrong) directly in Isaac Sim and save it as a usd file, and set the intrinsic parameters the same as RealSense D435i. Will this help me achieve the equivalent of adding a Realsense D435i camera? It seems that the robot can receive the rgb and depth value through get_obs() function, and there is also one Viewport from the camera, which is bound to the robotic wrist. 

Btw, I am also curious about what's the difference for the robot in OmniGibson between adding a Camera directly in Isaac Sim, saving it as a usd file and adding a VisionSensor in yaml file? 

Looking forward to your reply!

Best regards,

Yunfeng
